pallets/score-staking/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+					stake_coef * total_stake +
+						score_coef * <u32 as Into<ParaStaking::BalanceOf<T>>>::into(total_score),
+				) * round_reward;
+
+				p.unpaid += user_reward;
+				total_user_reward += user_reward;
+				Scores::<T>::insert(&a, p);
+				weight = weight.saturating_add(T::DbWeight::get().reads_writes(2, 1));
+			}
+
+			Self::deposit_event(Event::<T>::RewardCalculated {
+				total: round_reward,
+				distributed: total_user_reward,
+			});
+
+			weight
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight((2 * T::DbWeight::get().write, DispatchClass::Normal))]
+		pub fn set_score_feeder(
+			origin: OriginFor<T>,
+			new_score_feeder: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::AdminOrigin::ensure_origin(origin)?;
+			ScoreFeeder::<T>::put(new_score_feeder.clone());
+			Self::deposit_event(Event::ScoreFeederSet { new_score_feeder: Some(new_score_feeder) });
+			Ok(Pays::No.into())
+		}
+
+		/// Start (or restart) a currently stopped pool
+		///
+		/// It:
+		/// - sets the RoundInfo.start_block to the current block number
+		/// - advances the round index
+		#[pallet::call_index(1)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn start_pool(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			T::AdminOrigin::ensure_origin(origin)?;
+			ensure!(Self::state() == PoolState::Stopped, Error::<T>::PoolAlreadyRunning);
+			State::<T>::put(PoolState::Running);
+			let start_block = frame_system::Pallet::<T>::block_number();
+			let mut r = Round::<T>::take();
+			r.index = r.index.checked_add(1).ok_or(Error::<T>::RoundIndexOverflow)?;
+			r.start_block = start_block;
+			Round::<T>::put(r);
+			Self::deposit_event(Event::PoolStarted { start_block });
+			Ok(Pays::No.into())
+		}
+
+		/// Stop a currently running pool, should be called as caution
+		#[pallet::call_index(2)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn stop_pool(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			T::AdminOrigin::ensure_origin(origin)?;
+			ensure!(Self::state() == PoolState::Running, Error::<T>::PoolNotRun);
+			State::<T>::put(PoolState::Stopped);
+			Self::deposit_event(Event::PoolStopped {});
+			Ok(Pays::No.into())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn set_round_config(
+			origin: OriginFor<T>,
+			config: RoundSetting,
+		) -> DispatchResultWithPostInfo {
+			T::AdminOrigin::ensure_origin(origin)?;
+			RoundConfig::<T>::put(config);
+			Self::deposit_event(Event::RoundConfigSet { new_config: config });
+			Ok(Pays::No.into())
+		}
+
+		// Intentionally use `Identity` type to lower the hurdle of mapping to the
+		// desired substrate account as it's handled on-chain instead of by client.
+		//
+		// Subject to requirement change though
+		#[pallet::call_index(4)]
+		#[pallet::weight((195_000_000, DispatchClass::Normal))]
+		pub fn update_score(
```

Comment:
I think they will need a batch updating method too.

Reply:
they are free to use native `utility.batch`